src/routes/+page.svelte

```diff
@@ -1,2 +1,93 @@
-<h1>Welcome to SvelteKit</h1>
-<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
+<script>
+	import { createEventDispatcher } from 'svelte';
+
+	let url = '';
+	let shortUrl = '';
+	let error = '';
+	let loading = false;
+	let success = false;
+
+	const dispatch = createEventDispatcher();
+
+	function validateUrl(value) {
+		try {
+			new URL(value);
+			return true;
+		} catch (_) {
+			return false;
+		}
+	}
+
+	async function handleSubmit(event) {
+		event.preventDefault();
+		error = '';
+		success = false;
+
+		if (!validateUrl(url)) {
+			error = 'Invalid URL';
+			return;
+		}
+
+		loading = true;
+
+		try {
+			// Simulate an API call
+			await new Promise((resolve) => setTimeout(resolve, 1000));
+
+			// Check if shortUrl already exists (simulate)
+			if (shortUrl === 'existing') {
+				throw new Error('Short URL already exists');
+			}
+
+			success = true;
+			dispatch('success', { shortUrl });
+		} catch (err) {
+			error = err.message;
+		} finally {
+			loading = false;
+		}
+	}
+</script>
+
+<main class="container">
+	<h1>URL Shortener</h1>
+	<form on:submit={handleSubmit}>
```

Comment:
https://learn.svelte.dev/tutorial/customizing-use-enhance

Reply:
got it! thanks, ...rebuilding with `use:enchance`